### ISSUE TITLE
confirm success of upload

### ISSUE THREAD

**Issue by matkoniecz** [ACRA/acra#598]
it may be desirable to display toast confirming that report was successfully send (or that sending failed)

reported from downstream - https://github.com/MarcusWolschon/osmeditor4android/issues/605#issuecomment-332646948

**Comment by F43nd1r** [comment]
That option exists, if you are using dialog mode: use `resDialogOkToast`

**Comment by simonpoole** [comment]
resDialogOkToast only shows a confirmation when the user has submitted a report, independently of it could actually be submitted to the destination for such reports (I had a look at the code and there is no provision for notifying on successful/failed upload).

**Comment by F43nd1r** [comment]
I've added toast options to version 5.0.0-alpha10: https://github.com/F43nd1r/acra/blob/modules/acra-core/src/main/java/org/acra/annotation/AcraCore.java#L415

As this is in alpha stage, the changes needed to use it are not well documented, but if you post your configuration here, I'll transform it.

**Comment by simonpoole** [comment]
@F43nd1r glad to test as soon as I've worked through the current pile of hi-prio items, current config:

@ReportsCrashes(
	reportType = org.acra.sender.HttpSender.Type.JSON,
	httpMethod = org.acra.sender.HttpSender.Method.PUT,
	formUri = "http://acralyzer.vespucci.io/acraproxy",
	mode = ReportingInteractionMode.DIALOG,
	resDialogText = R.string.crash_dialog_text,
	resDialogCommentPrompt = R.string.crash_dialog_comment_prompt,
	resDialogTheme = R.style.Theme_AppCompat_Light_Dialog)

**Comment by F43nd1r** [comment]
```java
@AcraCore(resReportSendSuccessToast = /*new string resource, optional*/,
        resReportSendFailureToast = /*new string resource, optional*/)
@AcraHttpSender(
        reportType = HttpSender.Type.JSON,
        httpMethod = HttpSender.Method.POST,
        uri = "http://acralyzer.vespucci.io/acraproxy")
@AcraDialog(
        resText = R.string.crash_dialog_text,
        resCommentPrompt = R.string.crash_dialog_comment_prompt,
        resTheme = R.style.Theme_AppCompat_Light_Dialog)
```

and these are the required dependencies:
```groovy
 compile 'com.faendir.acra:acra-http:5.0.0-alpha10'
 compile 'com.faendir.acra:acra-dialog:5.0.0-alpha10'
```

**Comment by simonpoole** [comment]
@F43nd1r I've updated the dependency to 5.0.0 and changed the config, to
```
@AcraCore(resReportSendSuccessToast = R.string.report_success, resReportSendFailureToast = R.string.report_failure)
@AcraHttpSender(httpMethod = HttpSender.Method.POST, uri = "http://acralyzer.vespucci.io/acraproxy")
@AcraDialog(resText = R.string.crash_dialog_text, resCommentPrompt = R.string.crash_dialog_comment_prompt, resTheme = R.style.Theme_AppCompat_Light_Dialog)

```

Which works fine, except that it doesn't actually display the toasts after successfully or unsuccessfully  submitting a report.

**Comment by F43nd1r** [comment]
Can confirm this issue. Problem is the thread dies before the toast is sent.
Related stacktrace 
```
W/MessageQueue: Handler (android.widget.Toast$TN$1) {3862f92} sending message to a Handler on a dead thread
                                                                         java.lang.IllegalStateException: Handler (android.widget.Toast$TN$1) {3862f92} sending message to a Handler on a dead thread
                                                                             at android.os.MessageQueue.enqueueMessage(MessageQueue.java:545)
                                                                             at android.os.Handler.enqueueMessage(Handler.java:661)
                                                                             at android.os.Handler.sendMessageAtTime(Handler.java:630)
                                                                             at android.os.Handler.sendMessageDelayed(Handler.java:600)
                                                                             at android.os.Handler.sendMessage(Handler.java:537)
                                                                             at android.os.Message.sendToTarget(Message.java:418)
                                                                             at android.widget.Toast$TN.hide(Toast.java:441)
                                                                             at android.app.ITransientNotification$Stub.onTransact(ITransientNotification.java:57)
                                                                             at android.os.Binder.execTransact(Binder.java:674)

```

**Comment by F43nd1r** [comment]
Hopefully finally fixed in 5.0.2
